### ISSUE TITLE
Spec failure on ruby2.3

### ISSUE THREAD

**Issue by zeha** [berkshelf/varia_model#17]
Hi,

I'm trying to run specs with ruby 2.3.0 and get a few failures:

```
All examples were filtered out; ignoring {:focus=>true}
.....F...............FF...FF..FF..............FF..F.FF..F......F....

Failures:

  1) VariaModel ClassMethods ::attribute sets a default value if :default option is provided
     Failure/Error: expect(subject.attributes.dig('brooke.winsor')).to eql('rhode island')
       
       expected: "rhode island"
            got: nil
       
       (compared using eql?)
     # ./spec/unit/varia_model_spec.rb:44:in `block (4 levels) in <top (required)>'

  2) VariaModel ClassMethods ::validate_kind_of success returns an array where the first element is ':ok'
     Failure/Error: expect(subject.validate_kind_of(types, model, key).first).to eql(:ok)
       
       expected: :ok
            got: :error
       
       (compared using eql?)
       
       Diff:
       @@ -1,2 +1,2 @@
       -:ok
       +:error
     # ./spec/unit/varia_model_spec.rb:174:in `block (5 levels) in <top (required)>'

  3) VariaModel ClassMethods ::validate_kind_of success returns an array where the second element is a blank string
     Failure/Error: expect(subject.validate_kind_of(types, model, key)[1]).to be_blank
       expected `"Expected attribute: 'nested.one' to be a type of: 'String', 'Buff::Boolean'".blank?` to return true, got false
     # ./spec/unit/varia_model_spec.rb:178:in `block (5 levels) in <top (required)>'

  4) VariaModel ClassMethods ::validate_required passes validation if the value of the attribute is false
     Failure/Error: expect(subject.validate_required(model, key).first).to eql(:ok)
       
       expected: :ok
            got: :error
       
       (compared using eql?)
       
       Diff:
       @@ -1,2 +1,2 @@
       -:ok
       +:error
     # ./spec/unit/varia_model_spec.rb:229:in `block (4 levels) in <top (required)>'

  5) VariaModel ClassMethods ::validate_required passes validation if the value of the attribute is not nil
     Failure/Error: expect(subject.validate_required(model, key).first).to eql(:ok)
       
       expected: :ok
            got: :error
       
       (compared using eql?)
       
       Diff:
       @@ -1,2 +1,2 @@
       -:ok
       +:error
     # ./spec/unit/varia_model_spec.rb:235:in `block (4 levels) in <top (required)>'

  6) VariaModel ClassMethods ::validate_required success returns an array where the first element is ':ok'
     Failure/Error: expect(subject.validate_required(model, key).first).to eql(:ok)
       
       expected: :ok
            got: :error
       
       (compared using eql?)
       
       Diff:
       @@ -1,2 +1,2 @@
       -:ok
       +:error
     # ./spec/unit/varia_model_spec.rb:254:in `block (5 levels) in <top (required)>'

  7) VariaModel ClassMethods ::validate_required success returns an array where the second element is a blank string
     Failure/Error: expect(subject.validate_required(model, key)[1]).to be_blank
       expected `"A value is required for attribute: 'nested.one'".blank?` to return true, got false
     # ./spec/unit/varia_model_spec.rb:258:in `block (5 levels) in <top (required)>'

  8) VariaModel Validations validate type returns false if it fails validation
     Failure/Error: expect(subject).not_to be_valid
       expected `#<#<Class:0x0000000301f3d0>:0x0000000301eb88 @_attributes_=#<VariaModel::Attributes brooke=#<VariaModel::Attributes winsor=false>>, @errors=#<Hashie::Mash>>.valid?` to return false, got true
     # ./spec/unit/varia_model_spec.rb:425:in `block (4 levels) in <top (required)>'

  9) VariaModel Validations validate type adds an error if it fails validation
     Failure/Error: expect(subject.errors.size).to eq(1)
       
       expected: 1
            got: 0
       
       (compared using ==)
     # ./spec/unit/varia_model_spec.rb:431:in `block (4 levels) in <top (required)>'

  10) VariaModel#get_attribute returns the value of the given dotted path
      Failure/Error: expect(subject.get_attribute('brooke.winsor')).to eql('sister')
        
        expected: "sister"
             got: nil
        
        (compared using eql?)
      # ./spec/unit/varia_model_spec.rb:471:in `block (3 levels) in <top (required)>'

  11) VariaModel#get_attribute returns the current value of the Proc
      Failure/Error: expect(subject.get_attribute("brooke.winsor")).to eql("BACON")
        
        expected: "BACON"
             got: nil
        
        (compared using eql?)
      # ./spec/unit/varia_model_spec.rb:480:in `block (3 levels) in <top (required)>'

  12) VariaModel#get_attribute returns the current value of the Proc each time
      Failure/Error: expect(subject.get_attribute("brooke.winsor")).to eql("ponies")
        
        expected: "ponies"
             got: nil
        
        (compared using eql?)
      # ./spec/unit/varia_model_spec.rb:486:in `block (3 levels) in <top (required)>'

  13) VariaModel#mass_assign sets the values of all matching defined attributes
      Failure/Error: expect(subject.brooke.winsor).to eql("other")
        
        expected: "other"
             got: "sister"
        
        (compared using eql?)
      # ./spec/unit/varia_model_spec.rb:533:in `block (3 levels) in <top (required)>'

  14) VariaModel#from_hash updates and returns self from a Hash
      Failure/Error: expect(subject.first_name).to eql("jamie")
        
        expected: "jamie"
             got: nil
        
        (compared using eql?)
      # ./spec/unit/varia_model_spec.rb:620:in `block (3 levels) in <top (required)>'

Finished in 0.05469 seconds (files took 0.18868 seconds to load)
68 examples, 14 failures

Failed examples:

rspec ./spec/unit/varia_model_spec.rb:41 # VariaModel ClassMethods ::attribute sets a default value if :default option is provided
rspec ./spec/unit/varia_model_spec.rb:173 # VariaModel ClassMethods ::validate_kind_of success returns an array where the first element is ':ok'
rspec ./spec/unit/varia_model_spec.rb:177 # VariaModel ClassMethods ::validate_kind_of success returns an array where the second element is a blank string
rspec ./spec/unit/varia_model_spec.rb:226 # VariaModel ClassMethods ::validate_required passes validation if the value of the attribute is false
rspec ./spec/unit/varia_model_spec.rb:232 # VariaModel ClassMethods ::validate_required passes validation if the value of the attribute is not nil
rspec ./spec/unit/varia_model_spec.rb:253 # VariaModel ClassMethods ::validate_required success returns an array where the first element is ':ok'
rspec ./spec/unit/varia_model_spec.rb:257 # VariaModel ClassMethods ::validate_required success returns an array where the second element is a blank string
rspec ./spec/unit/varia_model_spec.rb:424 # VariaModel Validations validate type returns false if it fails validation
rspec ./spec/unit/varia_model_spec.rb:428 # VariaModel Validations validate type adds an error if it fails validation
rspec ./spec/unit/varia_model_spec.rb:470 # VariaModel#get_attribute returns the value of the given dotted path
rspec ./spec/unit/varia_model_spec.rb:478 # VariaModel#get_attribute returns the current value of the Proc
rspec ./spec/unit/varia_model_spec.rb:483 # VariaModel#get_attribute returns the current value of the Proc each time
rspec ./spec/unit/varia_model_spec.rb:522 # VariaModel#mass_assign sets the values of all matching defined attributes
rspec ./spec/unit/varia_model_spec.rb:617 # VariaModel#from_hash updates and returns self from a Hash
```

I think there were some changes to Hash in 2.3. Maybe you could take a look?

Thanks!

**Comment by gravitystorm** [comment]
I've investigated all of the failures, and they are reasonably easy to fix. `varia_model` uses the `Hash#dig` method from buff-extensions, and this clashes with `Hash#dig` from ruby 2.3.0. I've proposed renaming the method in https://github.com/reset/buff-extensions/pull/7 . With an updated version of buff-extensions, and changing all the 'dig' to 'dotted_dig' in the varia_model codebase, all the tests then pass on ruby 2.3.0.

Or in short, although there's 14 failures in the test suite, there's only one underlying problem and it's reasonably easy to fix.
